src/test/java/underdogs/devbie/notice/service/NoticeServiceTest.java

```diff
@@ -0,0 +1,168 @@
+import underdogs.devbie.notice.domain.Language;
+import underdogs.devbie.notice.domain.Notice;
+import underdogs.devbie.notice.domain.NoticeDescription;
+import underdogs.devbie.notice.domain.NoticeRepository;
+import underdogs.devbie.notice.dto.NoticeCreateRequest;
+import underdogs.devbie.notice.dto.NoticeDetailResponse;
+import underdogs.devbie.notice.dto.NoticeResponse;
+import underdogs.devbie.notice.dto.NoticeUpdateRequest;
+
+@ExtendWith(MockitoExtension.class)
+public class NoticeServiceTest {
+
+    private NoticeService noticeService;
+
+    @Mock
+    private NoticeRepository noticeRepository;
+
+    @BeforeEach
+    void setUp() {
+        noticeService = new NoticeService(noticeRepository);
+    }
+
+    @DisplayName("게시글 저장")
+    @Test
+    void save() {
+        Set<String> languages = Stream.of(Language.JAVA.getName(), Language.JAVASCRIPT.getName())
+            .collect(Collectors.toSet());
+        Notice expected = Notice.builder()
+            .id(1L)
+            .company(new Company("underdogs", 50_000_000))
+            .jobPosition(JobPosition.BACKEND)
+            .noticeDescription(new NoticeDescription(languages, "We are hiring!"))
+            .image("/static/image/underdogs")
+            .duration(new Duration(LocalDateTime.now(), LocalDateTime.now()))
+            .build();
+        given(noticeRepository.save(any(Notice.class))).willReturn(expected);
+
+        NoticeCreateRequest noticeRequest = NoticeCreateRequest.builder()
+            .name("underdogs")
+            .salary(50_000_000)
+            .languages(Arrays.asList(Language.JAVA.getName(), Language.JAVASCRIPT.getName()))
+            .jobPosition(JobPosition.BACKEND)
+            .image("/static/image/underdogs")
+            .description("We are hiring!")
+            .startDate(String.valueOf(LocalDateTime.now()))
+            .endDate(String.valueOf(LocalDateTime.now()))
+            .build();
+
+        Long noticeId = noticeService.save(noticeRequest);
+
+        assertThat(noticeId).isEqualTo(expected.getId());
+    }
+
+    @DisplayName("게시글 업데이트")
+    @Test
+    void update() {
+        Long id = 2L;
+        NoticeUpdateRequest request = NoticeUpdateRequest.builder()
+            .name("underdogs")
+            .salary(50_000_000)
+            .languages(Arrays.asList(Language.JAVA.getName(), Language.JAVASCRIPT.getName()))
+            .jobPosition(JobPosition.BACKEND)
+            .image("/static/image/underdogs")
+            .description("We are hiring!")
+            .startDate(String.valueOf(LocalDateTime.now()))
+            .endDate(String.valueOf(LocalDateTime.now()))
+            .build();
+
+        given(noticeRepository.findById(anyLong())).willReturn(Optional.of(request.toEntity(2L)));
+
+        noticeService.update(id, request);
+
+        verify(noticeRepository).findById(eq(2L));
+    }
+
+    @DisplayName("게시글 삭제")
+    @Test
+    void delete() {
+        doNothing().when(noticeRepository).deleteById(anyLong());
```

Comment:
BDD style을 쓰기로한만큼 여기도 통일성을 가져가면 어떨까요?

```suggestion
        willDoNothing().given(noticeRepository).deleteById(anyLong());
```